04_numeronym/joel00wood/main.go

```diff
@@ -0,0 +1,35 @@
+package main
+
+import (
+	"fmt"
+	"strconv"
+)
+
+func main() {
+	fmt.Println(numeronyms("accessibility", "Kubernetes", "abc"))
+}
+
+func numeronym(s string) string {
+	var out string
+	len := len(s)
+	midLen := len - 2
+	if midLen > 1 {
+		start := s[:1]
+		mid := strconv.Itoa(midLen)
+		end := s[len-1:]
+		out = start + mid + end
+	} else {
+		out = s
+	}
+
+	return out
+}
+
+func numeronyms(vals ...string) []string {
+	out := []string{}
+	for _, val := range vals {
+		nym := numeronym(val)
+		out = append(out, nym)
```

Comment:
FWIW: `append` is quite slow since it creates a whole new array every iteration

Since you know the length of the output array is the same as your input array you could just as easily do:

```go
out := make([]string, len(vals))
for idx, val := range values {
    out[idx] = numeronym(val)
}

return out
```

Reply:
This (`make()`) makes a sized slice, you could also just use a regular array

Reply:
Live example:

https://goplay.space/#ok7pqrqoCL_W

Reply:
Made the change as described

Reply:
Oh I wasn't trying to force that one one ye, was more just an "interesting" side note, but cool :)